linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -1552,58 +1553,27 @@ object Analyzer {
   private val getSuperclassMethodName =
     MethodName("getSuperclass", Nil, ClassRef(ClassClass))
 
-  private class WorkQueue(val ec: ExecutionContext) {
-    private val queue = new ConcurrentLinkedQueue[() => Unit]()
-    private val working = new AtomicBoolean(false)
+  private class WorkTracker(implicit val ec: ExecutionContext) {
     private val pending = new AtomicInteger(0)
     private val promise = Promise[Unit]()
 
-    def enqueue[T](fut: Future[T])(onSuccess: T => Unit): Unit = {
+    def track[T](fut: Future[T])(onSuccess: T => Unit): Unit = {
       val got = pending.incrementAndGet()
       assert(got > 0)
 
-      fut.onComplete {
-        case Success(r) =>
-          queue.add(() => onSuccess(r))
-          tryDoWork()
-
-        case Failure(t) =>
-          promise.tryFailure(t)
-      } (ec)
-    }
-
-    def join(): Future[Unit] = {
-      tryDoWork()
-      promise.future
+      fut.map(onSuccess).onComplete {
```

Comment:
I'm wondering if I should add a commit removing the continuation passing style. IIUC we wouldn't make any more futures, the code would just be more idiomatic.